### ISSUE TITLE
Problem with roundToNearestMinutes function

### ISSUE THREAD

**Issue by sparrowek** [date-fns/date-fns#3545]
There is an issue with the [roundToNearestMinutes function](https://date-fns.org/v2.30.0/docs/roundToNearestMinutes)

  ## The problem

  ### 💻 Code demonstrating the problem

  

  ```js
        const startTime = roundToNearestMinutes(new Date(2023, 10, 10, 12, 23, 0), {
            nearestTo: 5,
            roundingMethod: 'floor',
        });
  ```

  ### 🙁 Actual behavior

  

```
{
   startTime:
Fri Nov 10 2023 12:25:00 GMT+0100 (Central European Standard Time)
}
```

  ### 🙂 Expected behavior

  
I would expect it to return 12:20 in the floor mode

  ## Debug information

  - date-fns version: 2.30.0

**Comment by gustavopch** [comment]
And changing `floor` to `ceil` results in `2023-11-10T15:30:00.000Z`.

**Comment by gustavopch** [comment]
Seems like there are many other issues about this:
https://github.com/date-fns/date-fns/issues/3429
https://github.com/date-fns/date-fns/issues/3268
https://github.com/date-fns/date-fns/issues/3242
https://github.com/date-fns/date-fns/issues/3129

And also a PR:
https://github.com/date-fns/date-fns/pull/3467

**Comment by fturmel** [comment]
Fixed in v3.2.0!
